ca_municipalities/people.py

```diff
@@ -0,0 +1,162 @@
+
+import re
+
+LIST_PAGE = 'https://www.civicinfo.bc.ca/people'
+
+
+class CanadaMunicipalitiesPersonScraper(CSVScraper):
+    csv_url = 'https://docs.google.com/spreadsheets/d/e/2PACX-1vRrGXQy8qk16OhuTjlccoGB4jL5e8X1CEqRbg896ufLdh67DQk9nuGm-oufIT0HRMPEnwePw2HDx1Vj/pub?gid=0&single=true&output=csv'
+    other_names = {}
+    corrections = {}
+    organizations = {}
+    encoding = 'utf-8'
+
+    """
+    Returns whether the row should be imported. By default, skips empty rows
+    and rows in which a name component is "Vacant".
+    """
+    def is_valid_row(self, row):
+        empty = ('', 'Vacant')
+        if not any(row.values()):
+            return False
+        if 'first name' in row and 'last name' in row:
+            return row['last name'] not in empty and row['first name'] not in empty
+        return row['full name'] not in empty
+
+    def validateOrganizationClassification(self, organizationClassification):
+        if not organizationClassification:
+            organizationClassification = 'government'
+        else:
+            allowedClassifications = ['legislature', 'executive', 'upper', 'lower', 'party', 'committee', 'commission', 'corporation', 'agency', 'department', 'government']
+            if (organizationClassification not in allowedClassifications):
+                organizationClassification = 'government'
+
+        return organizationClassification
+
+    def scrape(self):
+        seat_numbers = defaultdict(lambda: defaultdict(int))
+
+        reader = self.csv_reader(self.csv_url, delimiter=self.delimiter, header=True, encoding=self.encoding, skip_rows=self.skip_rows)
+        reader.fieldnames = [self.header_converter(field) for field in reader.fieldnames]
+        for row in reader:
+
+            try:
+                if self.is_valid_row(row):
+                    for key, corrections in self.corrections.items():
+                        if not isinstance(corrections, dict):
+                            row[key] = corrections(row[key])
+                        elif row[key] in corrections:
+                            row[key] = corrections[row[key]]
+
+                    organizationClassification = row.get('classification')
+                    organizationClassification = self.validateOrganizationClassification(organizationClassification)
+
+                    organization = None
+                    organizationName = row['organization']
+                    if organizationName:
+                        if self.organizations.get(organizationName.lower()):
+                            organization = self.organizations.get(organizationName.lower())
+                        else:
+                            organization = Organization(organizationName, classification=organizationClassification)
+                            organization.add_source(self.csv_url)
+                            yield organization
+                            self.organizations[organizationName] = organization
```

Comment:
There was a bug, in that the key wasn't being set to `.lower()` in the assignment. This code has less repetition and therefore avoids that bug:

```python
organization_key = organizationName.lower()
if organization_key in self.organizations:
    organization = self.organizations[organization_key]
else:
    organization = Organization(organizationName, classification=organizationClassification)
    organization.add_source(self.csv_url)
    yield organization
    self.organizations[organization_key] = organization
```

Reply:
Nice catch. Thanks